From 445479a4a41776c7569fe8690e258a3099b3f081 Mon Sep 17 00:00:00 2001
From: Zhang Rui <rui.zhang@intel.com>
Date: Wed, 22 Apr 2020 15:26:07 +0800
Subject: [PATCH] ACPI: processor: idle: Allow probing on platforms with
 one ACPI C-state

commit 496121c02127e9c460b436244c38260b044cc45a upstream.

It is possible for ACPI _CST to return only one ACPI C-state, for
example, when deep cstate disabled in the BIOS.

And it is better for the acpi_idle driver to probe succesfully in
this case as well for consistency.

Signed-off-by: Zhang Rui <rui.zhang@intel.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
[Ankit: Backport to v4.19.x]
Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 drivers/acpi/processor_idle.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/acpi/processor_idle.c b/drivers/acpi/processor_idle.c
index d80010ac2..ca797a071 100644
--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -324,8 +324,7 @@ static int acpi_processor_get_power_info_cst(struct acpi_processor *pr)
 
 	cst = buffer.pointer;
 
-	/* There must be at least 2 elements */
-	if (!cst || (cst->type != ACPI_TYPE_PACKAGE) || cst->package.count < 2) {
+	if (!cst || (cst->type != ACPI_TYPE_PACKAGE) || !cst->package.count) {
 		pr_err("not enough elements in _CST\n");
 		ret = -EFAULT;
 		goto end;
@@ -652,8 +651,7 @@ static int acpi_processor_get_cstate_info(struct acpi_processor *pr)
 	for (i = 1; i < ACPI_PROCESSOR_MAX_POWER; i++) {
 		if (pr->power.states[i].valid) {
 			pr->power.count = i;
-			if (pr->power.states[i].type >= ACPI_STATE_C2)
-				pr->flags.power = 1;
+			pr->flags.power = 1;
 		}
 	}
 
-- 
2.23.1

